src/diffusers/models/attention.py

```diff
@@ -213,6 +238,53 @@ def forward(self, hidden_states, context=None):
         return hidden_states
 
 
+class MemoryEfficientCrossAttention(nn.Module):
+    def __init__(self, query_dim, context_dim=None, heads=8, dim_head=64, dropout=0.0):
+        super().__init__()
+        inner_dim = dim_head * heads
+        context_dim = default(context_dim, query_dim)
+
+        self.heads = heads
+        self.dim_head = dim_head
+
+        self.to_q = nn.Linear(query_dim, inner_dim, bias=False)
+        self.to_k = nn.Linear(context_dim, inner_dim, bias=False)
+        self.to_v = nn.Linear(context_dim, inner_dim, bias=False)
+
+        self.to_out = nn.Sequential(nn.Linear(inner_dim, query_dim), nn.Dropout(dropout))
+        self.attention_op: Optional[Any] = None
+
+    def forward(self, x, context=None, mask=None):
+        q = self.to_q(x)
+        context = default(context, x)
+        k = self.to_k(context)
+        v = self.to_v(context)
+
+        b, _, _ = q.shape
+        q, k, v = map(
+            lambda t: t.unsqueeze(3)
+            .reshape(b, t.shape[1], self.heads, self.dim_head)
+            .permute(0, 2, 1, 3)
+            .reshape(b * self.heads, t.shape[1], self.dim_head)
+            .contiguous(),
```

Comment:
Note that a more recent implementation in xFormers allows us to pass input tensors without having to perform the permutation + contiguous, making the code run even faster. @danthe3rd can give more details on the API

Reply:
That would be great, from which commit onward is this enable. 
I am currently unable to build xformers from source using main, nor able to use the conda install procedure as it fails at runtime for me. 
I will apply the other modifications first and once I get a running set up with the latest version I will try to add this as well. 

For info I am currently using this version of xformers `pip install git+https://github.com/facebookresearch/xformers@51dd119#egg=xformers`

Reply:
It was enabled in this commit: https://github.com/facebookresearch/xformers/commit/7e4c02c455715635a9899c3d9b47508d9d7998c3

> I am currently unable to build xformers from source using main, nor able to use the conda install procedure as it fails at runtime for me.

Would it be possible for you to open an issue in xformers? Our setup procedure is fairly new and we would like to make it as easy as possible to get things running - either when building it locally, or with conda packages

Reply:
Yes will do !